flexget/components/ftp/sftp_client.py

```diff
@@ -26,38 +26,230 @@
-        private_key: Optional[str] = None,
-        private_key_pass: Optional[str] = None,
-        connection_tries: int = 3,
+        sftp: 'pysftp.Connection',
+        url_prefix: str,
+        private_key: Optional[str],
+        private_key_pass: Optional[str],
     ):
+        self._sftp = sftp
+        self._prefix = url_prefix
+        self._private_key = private_key
+        self._private_key_pass = private_key_pass
 
-        if not pysftp:
-            raise plugin.DependencyError(
-                issued_by='sftp_client',
-                missing='pysftp',
-                message='sftp client requires the pysftp Python module.',
-            )
+    def get_file_handler(self, get_size: bool, entry_accumulator: list) -> NodeHandler:
+        """
+        Builds a file node handler suitable for use with pysftp.Connection.walktree
 
-        self.host: str = host
-        self.port: int = port
-        self.username: str = username
-        self.password: Optional[str] = password
-        self.private_key: Optional[str] = private_key
-        self.private_key_pass: Optional[str] = private_key_pass
-
-        self.prefix: str = self._get_prefix()
-        self._sftp: 'pysftp.Connection' = self._connect(connection_tries)
-        self._handler_builder: HandlerBuilder = HandlerBuilder(
-            self._sftp, self.prefix, self.private_key, self.private_key_pass
+        :param get_size: boolean indicating whether to compute the for each file
+        :param entry_accumulator: list to add entries to
+        """
+        return partial(
+            Handlers.handle_file,
+            self._sftp,
+            self._prefix,
+            get_size,
+            self._private_key,
+            self._private_key_pass,
+            entry_accumulator,
         )
 
+    def get_dir_handler(
+        self, get_size: bool, files_only: bool, entry_accumulator: list
+    ) -> NodeHandler:
+        """
+        Builds a file node handler suitable for use with pysftp.Connection.walktree
+
+        :param get_size: boolean indicating whether to compute the for each file
+        :param files_only: Boolean indicating whether to skip directories
+        :param entry_accumulator: list to add entries to
+        """
+        return partial(
+            Handlers.handle_directory,
+            self._sftp,
+            self._prefix,
+            get_size,
+            files_only,
+            self._private_key,
+            self._private_key_pass,
+            entry_accumulator,
+        )
+
+    def get_unknown_handler(self) -> NodeHandler:
+        """
+        Builds an unknown node handler suitable for use with pysftp.Connection.walktree
+        """
+        return partial(Handlers.handle_unknown)
+
+    def get_null_handler(self) -> NodeHandler:
+        """
+        Builds a noop node handler suitable for use with pysftp.Connection.walktree
+        """
+        return partial(Handlers.null_node_handler)
+
+
+class Handlers:
+    @classmethod
```

Comment:
Complicated use of classmethods, they are generally alternative constructors.

Reply:
I basically want to use them like static methods, but with access to the other methods in the class. Would it be preferable to fully qualify them instead?